src/random.cr

```diff
@@ -1,66 +1,125 @@
 require "random/mt19937"
 
+# `Random` provides an interface for random values generation, using a pseudo random number generator (PRNG).
+# 
+# ```
+# Random.new_seed # => 112705036
+# Random.rand     # => 0.167595
+# Random.rand(5)  # => 2
+# ```
+#
+# The above methods delegate to a shared instance. You can create one like:
+#
+# ```
+# r = Random.new
+# r.next_bool # => true
+# r.next_int  # => 2223112
+# ```
+#
+# It also defines a global method `#rand`, which `Array#sample` delegates.
+#
+# ```
+# rand     # => 0.293829
+# rand(10) # => 8
+# ```
 module Random
   DEFAULT = MT19937.new
 
+  # Generates a `UInt32` based on current time using `read_cycle_counter` from LLVM.
   def self.new_seed : UInt32
     Intrinsics.read_cycle_counter.to_u32
   end
 
+  # Initiates an instance with the given *seed*. (Default: `#new_seed`)
   def self.new(seed = new_seed)
     MT19937.new(seed)
   end
 
+  # Generates a random `UInt32`.
   abstract def next_u32
 
+  # Generates a random `Bool`.
+  #
+  # ```
+  # Random.new.next_bool # => true
+  # ```
   def next_bool : Bool
     next_u32.even?
   end
 
+  # Generates a random `Int32`.
+  #
+  # ```
+  # Random.new.next_int # => 440038499
+  # Random.new.next_int # => -1848788484
+  # ```
   def next_int : Int32
     next_u32.to_i32
   end
 
+  # see `#rand`
   def next_float : Float64
     # Divided by 2^32-1
     next_u32 * (1.0/4294967295.0)
   end
 
+  # Generates a random `Float64` between 0 and 1.
+  #
+  # ```
+  # r = Random.new
+  # r.rand # => 0.167595
+  # r.rand # => 0.0372991
+  # ```
   def rand : Float64
     next_float
   end
 
-  def rand(x : Int) : Int32
-    if x > 0
-      (next_u32 % x).to_i32
+  # Returns a random `Int32` between 0 and *max*.
```

Comment:
You should put "greater than 0 and less than *max*" to emphasize the value will always be less than the max.